csm/core/services/file_transfer.py

```diff
@@ -89,7 +89,21 @@ def get_file_path(self) -> str:
         return path_to_cached_file
 
     def save_file(self, dir_to_save, filename, overwrite=False):
-        Log.debug(f"Saving f{filename} at f{dir_to_save}")
+        if not os.path.exists(dir_to_save):
+            Log.warn(f"{dir_to_save} not found. Recreating storage directory." )
+            try:
+                original_mask = os.umask(0o007)
+                Log.debug(f"original mask: {original_mask}")
+                os.makedirs(dir_to_save)
+            except Exception as e:
+                Log.debug(f"Can not create directory {e}")
```

Comment:
This should be warning or error 
Log.warn(f"Can not create directory {e}")

Reply:
Comment Addressed.